ghcide/src/Development/IDE/Core/Compile.hs

```diff
@@ -998,20 +998,11 @@ getDocsBatch
   :: HscEnv
   -> Module  -- ^ a moudle where the names are in scope
   -> [Name]
-  -> IO (Either ErrorMessages (Map.Map Name (Either T.Text (Maybe HsDocString, Maybe (Map.Map Int HsDocString)))))
+  -> IO (Either ErrorMessages (Map.Map Name (Either GetDocsFailure (Maybe HsDocString, Maybe (Map.Map Int HsDocString)))))
 getDocsBatch hsc_env _mod _names = do
     ((_warns,errs), res) <- initTc hsc_env HsSrcFile False _mod fakeSpan $ Map.fromList <$> traverse findNameInfo _names
-    pure $ case res of
-        Just x  -> pure $ fun x
-        Nothing -> Left errs
```

Comment:
This was just mapping of `Maybe` constructors to semantically according `Either` constructors.

Since `fun1` type conversion no longer happens - type transformation with the function no longer needed.